src/client/opamTreeCommand.ml

```diff
@@ -399,24 +402,25 @@ let simulate_new_state tog st universe install names =
          (OpamSwitchState.unavailable_reason st) cs);
     OpamStd.Sys.exit_because `No_solution
 
-let dry_install tog st universe install =
-  simulate_new_state tog st universe install
-    (OpamPackage.Name.Set.of_list (List.map fst install))
-
 let run st tog ?no_constraint mode filter atoms =
   let open OpamPackage.Set.Op in
-  let select, missing =
-    List.fold_left (fun (select, missing) atom ->
+  let select, missing, requested =
+    List.fold_left (fun (select, missing, requested) atom ->
         let installed =
           OpamPackage.Set.filter (OpamFormula.check atom) st.installed
         in
         if OpamPackage.Set.is_empty installed then
-          (select, atom :: missing)
-        else (installed ++ select, missing))
-      (OpamPackage.Set.empty, []) atoms
+          let missing_pkgs = OpamPackage.Set.filter (OpamFormula.check atom) st.packages in
+          if OpamPackage.Set.is_empty missing_pkgs then
+            assert false;
```

Comment:
Why this assert false ? it should be possible to query a non existing package, and it will fail when it tries to dry install it.

Reply:
Ah yes, my bad, I completely forgot about non-existing packages. Removed in https://github.com/ocaml/opam/pull/5919/commits/36cd0528f8f704489e8b215834cda738007295e1